doc/sphinx/proof-engine/ltac.rst

```diff
@@ -1160,6 +1173,388 @@ Printing |Ltac| tactics
+       | perm_trans : forall l0 l1 l2, perm l0 l1 -> perm l1 l2 -> perm l0 l2.
+
+.. coqtop:: in
+
+   End Sort.
+
+First, we define the permutation predicate as shown above.
+
+.. coqtop:: none
+
+   Require Import List.
+
+
+.. coqtop:: in
+
+   Ltac perm_aux n :=
+   match goal with
+       | |- (perm _ ?l ?l) => apply perm_refl
+       | |- (perm _ (?a :: ?l1) (?a :: ?l2)) =>
+           let newn := eval compute in (length l1) in
+               (apply perm_cons; perm_aux newn)
+       | |- (perm ?A (?a :: ?l1) ?l2) =>
+           match eval compute in n with
+               | 1 => fail
+               | _ =>
+                   let l1' := constr:(l1 ++ a :: nil) in
+                       (apply (perm_trans A (a :: l1) l1' l2);
+                       [ apply perm_append | compute; perm_aux (pred n) ])
+           end
+   end.
+
+Next we define an auxiliary tactic ``perm_aux`` which takes an argument
+used to control the recursion depth. This tactic behaves as follows. If
+the lists are identical (i.e. convertible), it concludes. Otherwise, if
+the lists have identical heads, it proceeds to look at their tails.
+Finally, if the lists have different heads, it rotates the first list by
+putting its head at the end if the new head hasn't been the head previously. To check this, we keep track of the
+number of performed rotations using the argument ``n``. We do this by
+decrementing ``n`` each time we perform a rotation. It works because
+for a list of length ``n`` we can make exactly ``n - 1`` rotations
+to generate at most ``n`` distinct lists. Notice that we use the natural
+numbers of Coq for the rotation counter. From :ref:`ltac-syntax` we know
+that it is possible to use the usual natural numbers, but they are only
+used as arguments for primitive tactics and they cannot be handled, so,
+in particular, we cannot make computations with them. Thus the natural
+choice is to use Coq data structures so that Coq makes the computations
+(reductions) by ``eval compute in`` and we can get the terms back by match.
+
+.. coqtop:: in
+
+   Ltac solve_perm :=
+   match goal with
+       | |- (perm _ ?l1 ?l2) =>
+           match eval compute in (length l1 = length l2) with
+               | (?n = ?n) => perm_aux n
+           end
+   end.
+
+The main tactic is ``solve_perm``. It computes the lengths of the two lists
+and uses them as arguments to call ``perm_aux`` if the lengths are equal (if they
+aren't, the lists cannot be permutations of each other). Using this tactic we
+can now prove lemmas as follows:
+
+.. coqtop:: in
+
+   Lemma solve_perm_ex1 :
+     perm nat (1 :: 2 :: 3 :: nil) (3 :: 2 :: 1 :: nil).
+   Proof. solve_perm. Qed.
+
+.. coqtop:: in
+
+   Lemma solve_perm_ex2 :
+     perm nat
+       (0 :: 1 :: 2 :: 3 :: 4 :: 5 :: 6 :: 7 :: 8 :: 9 :: nil)
+         (0 :: 2 :: 4 :: 6 :: 8 :: 9 :: 7 :: 5 :: 3 :: 1 :: nil).
+   Proof. solve_perm. Qed.
+
+Deciding intuitionistic propositional logic
```

Comment:
Example box?

The title leads me think this will tell me whether a given intuitionistic proposition is true or false/unprovable.  That's a proof, that's more of an algorithm.  What's a more accurate title?